### PR TITLE
Add agent hostNetwork option

### LINKED ISSUE #676: Agent daemonset should be started with hostNetwork
When using the `daemonset` stragegy for the agent, the agent pods should be created with `hostNetwork: true` to make it possible for tracing clients to reference their IP using the following container spec:
```
env
- name: JAEGER_AGENT_HOST
          valueFrom:
            fieldRef:
              fieldPath: status.hostIP
```

`hostNetwork` is enabled in the kubernetes production template too, so this looks like a small  oversight: https://github.com/jaegertracing/jaeger-kubernetes/blob/master/jaeger-production-template.yml#L217

### PR DESCRIPTION
**Pull request by faceair**
Add the `hostNetwork` option to the agent DaemonSet.

Fixes #676.

### CODE REVIEW

**Comment by faceair** [comment]
I can't get past the `make security` check, and now the code of the master branch is broken. For example, the type of the [clientset](https://github.com/jaegertracing/jaeger-operator/blob/v1.20.0/pkg/client/versioned/clientset.go#L17
) reference doesn't exist, and the [generated directory](https://github.com/jaegertracing/jaeger-operator/tree/v1.20.0/pkg/client/versioned/typed/jaegertracing/v1) is empty.

I'm confused as to why other people's PRs can do the build.

**Comment by jpkrohling** [comment]
> and now the code of the master branch is broken

It shouldn't be broken, CI is working for it, at least for the last-1 commit. Are you unable to build master locally? If so, you might have different tools/setup than what's expected.

**Comment by faceair** [comment]
Now here's why my CI build is failing: https://github.com/jaegertracing/jaeger-operator/pull/1257/checks?check_run_id=1253291006
My local `make ci` failed for the same reason.

**Comment by jpkrohling** [comment]
Are you unable to build master locally?

**Comment by faceair** [comment]
Yes, I can't build master locally, but it looks like it's for a different reason.

```
> go version
go version go1.15.2 darwin/amd64

> operator-sdk version
operator-sdk version: "v0.18.2", commit: "f059b5e17447b0bbcef50846859519340c17ffad", kubernetes version: "v1.18.2", go version: "go1.13.10 darwin/amd64"

> git log -1
commit 5271b85d99f2861a5e2ae69d4b466feb0b8bd63f (HEAD -> master, origin/master, origin/HEAD)
Author: Kevin Earls <kearls@redhat.com>
Date:   Tue Oct 13 16:13:22 2020 +0200

    Add tests for OTEL ingester (#1252)

    Signed-off-by: Kevin Earls <kearls@redhat.com>

> make ci
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [jaegertracing:[v1], kafka:[v1beta1], ]
F1014 20:52:49.984684    3973 deepcopy.go:922] Hit an unsupported type invalid type for *invalid type
Failed to generate the Kubernetes stubs.
make: *** [internal-generate] Error 255

> # after add GOROOT in makefile

> make ci
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [jaegertracing:[v1], kafka:[v1beta1], ]
INFO[0037] Code-generation complete.
Formatting code...
pkg/client/versioned/clientset.go
pkg/client/versioned/fake/clientset_generated.go
pkg/client/versioned/fake/register.go
pkg/client/versioned/scheme/register.go
Build failed: the versioned clients aren't up to date. Run 'make generate'.
make: *** [ensure-generate-is-noop] Error 1

> make generate
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [jaegertracing:[v1], kafka:[v1beta1], ]
INFO[0033] Code-generation complete.
Formatting code...
pkg/client/versioned/clientset.go
pkg/client/versioned/fake/clientset_generated.go
pkg/client/versioned/fake/register.go
pkg/client/versioned/scheme/register.go

> make ci                                                                                            1.3m
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [jaegertracing:[v1], kafka:[v1beta1], ]
INFO[0034] Code-generation complete.
Formatting code...
pkg/client/versioned/clientset.go
pkg/client/versioned/fake/clientset_generated.go
pkg/client/versioned/fake/register.go
pkg/client/versioned/scheme/register.go
Build failed: the versioned clients aren't up to date. Run 'make generate'.
make: *** [ensure-generate-is-noop] Error 1

**Comment by jpkrohling** [comment]
Then there's certainly something odd with your setup, as master indeed works in CI:

image

**Comment by faceair** [comment]
I found that the version of gosec used in [CI in the latest commit](https://pipelines.actions.githubusercontent.com/rWD2BC5Jx5QlkJonRBT4exuILu2P4UBZRU6AEUBXCpT2ZFzgse/_apis/pipelines/1/runs/4914/signedlogcontent/3?urlExpires=2020-10-14T18%3A27%3A21.5318450Z&urlSigningMethod=HMACV1&urlSignature=MiHdmcUFy4tQ0xJ08kOuJvDxINuCz%2BAffKCrqHlZTbQ%3D) is `v0.0.0-20191008095658-28c1128b7336`
And the version of gosec [my PR's](https://pipelines.actions.githubusercontent.com/rWD2BC5Jx5QlkJonRBT4exuILu2P4UBZRU6AEUBXCpT2ZFzgse/_apis/pipelines/1/runs/4899/signedlogcontent/3?urlExpires=2020-10-14T18%3A29%3A26.8777631Z&urlSigningMethod=HMACV1&urlSignature=pgGAB%2FMu53I7UyXmFpLNoZayAkt6XVzbj9FDoYL7Hlg%3D) CI uses is `v0.0.0-20200401082031-e946c8c39989`
Same script, getting a different version of the tool.

I now have gosec locked to `v0.0.0-20191008095658-28c1128b7336`. But it looks like I'm having other problems, so I'll keep working on that.

**Comment by faceair** [comment]
It looks like some of the issues are related to the fact that `go get` will modify the project's go.mod file, and I isolated the tools dependency via `vogt.sh`.

Hi @jpkrohling , please help review if there are any other issues!

**Comment by jpkrohling** [comment]
That's wonderful! The `vgot` script looks nice and might be interesting to @ibraimgm as well.

Would you be able to split this PR into two, one for each feature this one here is bringing? 

1. fix the dependencies (vgot changes + GOROOT)
1. add the hostNetwork. You'll also need to add a unit test for it.

File: pkg/apis/jaegertracing/v1/jaeger_types.go [hunk 1]
```diff
@@ -183,6 +183,9 @@ type JaegerCommonSpec struct {
 
 	// +optional
 	ServiceAccount string `json:"serviceAccount,omitempty"`
+
+	// +optional
+	HostNetWork *bool `json:"hostNetWork,omitempty"`
```
Comment (objectiser) [inline pkg/apis/jaegertracing/v1/jaeger_types.go:188]: @jpkrohling If this is only applicable to agent (currently), shouldn't it only be added to the `JaegerAgentSpec`?

Comment (faceair) [reply pkg/apis/jaegertracing/v1/jaeger_types.go:188]: Currently I'm only using it on the agent, but I'm not sure if that will change in the future. If you expect to only add to the JaegerAgentSpec, I can change that too.

Comment (objectiser) [reply pkg/apis/jaegertracing/v1/jaeger_types.go:188]: If you could add to the agent spec for now - and if a requirement arises in the future, we can move it to the common spec.

Comment (faceair) [reply pkg/apis/jaegertracing/v1/jaeger_types.go:188]: ok

**Review by jpkrohling — APPROVED** [review]

**Comment by faceair** [comment]
Hi @jpkrohling 
What is the planned release of jaeger-operator? How soon can I use this feature?

**Comment by jpkrohling** [comment]
We typically release new versions of the operator following new versions of Jaeger, which is typically released once every two months. The last release was about 20 days ago.

**Pull request merged**